### ISSUE TITLE
Addressing the problem of community

### ISSUE THREAD

**Issue by aaronduino** [upend/IF_MS_BUYS_GITHUB_IMMA_OUT#34]
While alternatives such as gitlab, gitea, gogs, etc, are great for internal projects where the authors rightly want to be in control of their own data, they (at least for the time being) lack the community that drives many to GitHub.

More specifically, the following may need to be addressed:

- stars. This positive feedback is a big motivator for many of my open source developer friends.
- visibility. Very few Show HNs are projects hosted on gitlab. Self-hosted repos are harder to find online.
- professional-ness. It is commonplace (afaik) to share one's GitHub profile page with peers and recuiters.

This is by no means a complete list. Feel free to list more things and they come to mind.

**Comment by bashrc** [comment]
I see visibility as being the main issue. You can always give a repo URL to peers or recruiters. It doesn't have to be Github.

I'll be thinking about ways to publish project summaries to some kind of p2p directory. FSF has its software directory, but I think something easier is needed.

**Comment by Bystroushaak** [comment]
I learned today that gitlab has its own section for hosted projects just like github. it just isn't much visible; https://gitlab.com/explore

**Comment by bashrc** [comment]
But unfortunately gitlab.com is a similar kind of deal to Github. Venture backed and presumably will be bought at some point, probably far more cheaply than $7.5 billion. It also runs on Microsoft Azure.

**Comment by Bystroushaak** [comment]
It is, but at least you can self-host it. And also there are propositions to make it federated:

* https://gitlab.com/gitlab-org/gitlab-ce/issues/4013

I think there will be much more pressure to implement it now.

**Comment by Sphinxs** [comment]
Ward.

**Comment by cmatzenbach** [comment]
@Bystroushaak That is true, but it's not the issue here.  People are missing the point.  This isn't (or rather, shouldn't) be some #deletefacebook thing about privacy - this is about GitHub, a company that just got sold for $7.5 billion solely because it has been the choice hub of the open source movement these past ten years, selling the platform to a company who has been the biggest enemy to open source software.  And no, the past five years don't make up for the past decades they've spent fighting FOSS tooth and nail.  They were losing out as web dev exploded and no one wanted to pay for their enterprise tools, so to stay relevant they opened up to linux and open source.  Then they realized that by doing so they can keep people on their proprietary platform and continue to make money, so they expanded things like WSL.  They're not doing it for the good of the community, it's all for profit.  Which is fine, that's the point of a company!  BUT that wasn't the point of GitHub, and that's certainly not why it became so popular.  We built it, and they sold it to the very people who have been trying to destroy this movement since the late 90s.

**Comment by nukeop** [comment]
The only thing that changed in the past 5 years is that they started seeing "open source" developers as a source of free labor that can be easily exploited. Many people contribute to VS Code for example, but all credit goes to Microsoft, it continues to be seen as a Microsoft project, and Microsoft is being praised for its development. They never license their work under GPL, they use MIT, or other licenses that make the projects useless for anyone that isn't Microsoft if they can get away with it.

**Comment by vassudanagunta** [comment]
Hey guys. I love the work you've been doing. Not everyone loves mine. Please read Issue 114.

**Comment by vassudanagunta** [comment]
Hello,

The GitHub Evacuation Project has moved to GitLab (not an endorsement or even a final home). Your enthusiasm and contribution is still needed. Please check out [the new project home](https://gitlab.com/upend/github), and read [the project wiki](https://gitlab.com/upend/github/evac-guide/wikis) for info to get restarted.

Thank you!

vas
